### PR TITLE
FIX: Reload values of the specified entity when change entity in the execution of cron jobs

### PR DESCRIPTION
**Pull request by kkhelifa-opendsi**
Reload values of the specified entity when change entity in the execution of cron jobs

### CODE REVIEW

File: htdocs/core/class/conf.class.php [hunk 1]
```diff
@@ -161,6 +174,58 @@ public function __construct()
 	 */
 	public function setValues($db)
 	{
+		// Unset all old modules values
+		if (!empty($this->modules)) {
```
Comment (eldy) [inline htdocs/core/class/conf.class.php:178]: The whole $this->modules is completely erased few lines after with
$this->modules = array();
So i think we can remove this portion of code, doesn't it ?

File: htdocs/core/class/conf.class.php [hunk 2]
```diff
@@ -161,6 +174,58 @@ public function __construct()
...
+			foreach ($this->modules as $m) {
+				if (isset($this->$m)) unset($this->$m);
+			}
+		}
+
+		// Properly declare multi-modules objects.
```
Comment (eldy) [inline htdocs/core/class/conf.class.php:184]: According to code comment and existing code, it seems that only 
		$this->file = new stdClass();
		$this->db = new stdClass();
are require to be set before the setValues is called.

So we should retrieve here all what is initialized into the constructor (except ->file and ->db).

What if you also add this ? Is this working on your side ?
		$this->mycompany = new stdClass();
		$this->admin = new stdClass();
		$this->user	= new stdClass();
		$this->browser = new stdClass();
		$this->medias = new stdClass();
		$this->cache = array();

**Comment by philippe-opendsi** [comment]
ping @aspangaro

**Comment by aspangaro** [comment]
Ping @eldy

**Pull request merged**